### PR TITLE
crtm: add noaa versions and package mods

### PR DESCRIPTION
**Pull request by eugeneswalker**
@AlexanderRichert-NOAA @BenjaminTJohnson @Hang-Lei-NOAA @climbfuji @edwardhartnett @t-brown

### CODE REVIEW

**Comment by climbfuji** [comment]
@eugeneswalker is this PR making the spack develop version identical to our fork version?

**Comment by eugeneswalker** [comment]
> @eugeneswalker is this PR making the spack develop version identical to our fork version?

I'm not sure this exactly matches the current state of that fork, but yes, these changes are coming from there.

**Review by AlexanderRichert-NOAA — CHANGES_REQUESTED** [review]
Please compare this against the jcsda version and note the differences. In particular, the netcdf-fortran dependency is not there for v2.4.1-jedi-- I'm assuming that needs to be there?

**Comment by eugeneswalker** [comment]
> Please compare this against the jcsda version and note the differences. In particular, the netcdf-fortran dependency is not there for v2.4.1-jedi-- I'm assuming that needs to be there?

Just added the missing depends_on for netcdf-fortran and ecbuild for crtm@2.4.1-jedi.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Review by climbfuji — CHANGES_REQUESTED** [review]
Can you please make this file identical with what's in our fork? This would be greatly appreciated. Otherwise someone will have to go and make another update right after your PR is merged. Thanks!

image

**Comment by eugeneswalker** [comment]
> Can you please make this file identical with what's in our fork? This would be greatly appreciated. Otherwise someone will have to go and make another update right after your PR is merged. Thanks!

I left these two lines out because there is no `@v3.0.0-rc.1` version defined, so it's not clear to me what's going on here:
1. https://github.com/JCSDA/spack/blob/1daf6f1d5df35d80d68ac742c9395dfb76997872/var/spack/repos/builtin/packages/crtm/package.py#L37
2. https://github.com/JCSDA/spack/blob/1daf6f1d5df35d80d68ac742c9395dfb76997872/var/spack/repos/builtin/packages/crtm/package.py#L47

I left the line below out because crtm-fix would also then need to be updated to include the version `@2.4.0.1_emc` which is not present currently, and this update would include the additional logic for handling that build which I don't follow:
1. https://github.com/JCSDA/spack/blob/1daf6f1d5df35d80d68ac742c9395dfb76997872/var/spack/repos/builtin/packages/crtm/package.py#L41

I don't have the bandwidth to track this down and test it out now. If you want to push to this PR please feel free, or we could do it in a separate PR or else delay it until I have more time in future.

**Comment by climbfuji** [comment]
> > Can you please make this file identical with what's in our fork? This would be greatly appreciated. Otherwise someone will have to go and make another update right after your PR is merged. Thanks!
> 
> I left these two lines out because there is no `@v3.0.0-rc.1` version defined, so it's not clear to me what's going on here:
> 
> 1. https://github.com/JCSDA/spack/blob/1daf6f1d5df35d80d68ac742c9395dfb76997872/var/spack/repos/builtin/packages/crtm/package.py#L37
> 2. https://github.com/JCSDA/spack/blob/1daf6f1d5df35d80d68ac742c9395dfb76997872/var/spack/repos/builtin/packages/crtm/package.py#L47
> 
> I left the line below out because crtm-fix would also then need to be updated to include the version `@2.4.0.1_emc` which is not present currently, and this update would include the additional logic for handling that build which I don't follow:
> 
> 1. https://github.com/JCSDA/spack/blob/1daf6f1d5df35d80d68ac742c9395dfb76997872/var/spack/repos/builtin/packages/crtm/package.py#L41
> 
> I don't have the bandwidth to track this down and test it out now. If you want to push to this PR please feel free, or we could do it in a separate PR or else delay it until I have more time in future.

Fair point. Let's merge this and I'll look at the rest later. Thanks!

**Review by climbfuji — APPROVED** [review]

File: var/spack/repos/builtin/packages/crtm/package.py [hunk 1]
```diff
@@ -34,20 +34,26 @@ class Crtm(CMakePackage):
...
 
     # ecbuild release v2.4.0 is broken
     # add ecbuild dependency for next release with fix
     # depends_on("ecbuild", when="@2.4.0:", type=("build"))
 
     license("CC0-1.0")
 
+    version(
+        "v2.4.1-jedi", sha256="fd8bf4db4f2a3b420b4186de84483ba2a36660519dffcb1e0ff14bfe8c6f6a14"
+    )
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/crtm/package.py:55]: I can't seem to confirm this version sha256 with `spack checksum`.  How was it calculated?

File: var/spack/repos/builtin/packages/crtm/package.py [hunk 2]
```diff
@@ -58,3 +64,17 @@ class Crtm(CMakePackage):
     # Branch release/crtm_jedi_v2.4.0
     version("v2.4-jedi.1", commit="82223419fdb479d76c2f2109c2b704e1d9618f22")
     version("v2.4-jedi.2", commit="62831cbb6c1ffcbb219eeec60e1b1c422526f597")
+
+    def url_for_version(self, version):
+        if self.spec.satisfies("@v3") or version >= Version("3.0.0"):
+            return "https://github.com/JCSDA/crtmv3/archive/refs/tags/{}.tar.gz".format(version)
+        else:
+            return "https://github.com/JCSDA/crtm/archive/refs/tags/{}.tar.gz".format(version)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/crtm/package.py:72]: FYI. We do encourage use of Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) when feasible, which we're moving to in the core.

**Review by tldahlgren — APPROVED** [review]
Confirmed the commit and version sha256.

**Pull request merged**